GVFS/GVFS.Common/FileBasedDictionary.cs

```diff
@@ -11,14 +11,28 @@ public class FileBasedDictionary<TKey, TValue> : FileBasedCollection
     {
         private ConcurrentDictionary<TKey, TValue> data = new ConcurrentDictionary<TKey, TValue>();
 
-        private FileBasedDictionary(ITracer tracer, PhysicalFileSystem fileSystem, string dataFilePath) 
+        private FileBasedDictionary(
+            ITracer tracer,
+            PhysicalFileSystem fileSystem, 
+            string dataFilePath, 
+            IEqualityComparer<TKey> keyComparer = null) 
             : base(tracer, fileSystem, dataFilePath, collectionAppendsDirectlyToFile: false)
         {
+            if (keyComparer != null)
+            {
+                this.data = new ConcurrentDictionary<TKey, TValue>(keyComparer);

```

Comment:
With this change `this.data` will be initialized twice.  Should we remove the creation of the `ConcurrentDictionary` here:

```private ConcurrentDictionary<TKey, TValue> data = new ConcurrentDictionary<TKey, TValue>();```

And always create it once in the `FileBasedDictionary` constructor?